src/pytest_allure_spec_coverage/models/collector.py

```diff
@@ -0,0 +1,51 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Abstract collector"""
+import os
+from abc import ABC, abstractmethod
+from typing import Optional, List
+
+import toml
+
+from pytest_allure_spec_coverage.models.scenario import Scenario
+
+
+class Collector(ABC):
+    """
+    Abstract collector
+    """
+
+    config: Optional[dict] = None
```

Comment:
Can we implement `Collector` `config` field as some class object (for example, `CollectorConfig`)?  According to the current field type, we don't know about the required and optional config params, as well as what params can be, because it is `dict`

Reply:
Config params will be different for each collector. Theoretically, we can use something like `Pydantic` to validate config, but maybe right now it's superfluous